packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -444,6 +445,127 @@ describe("ControllerHelper: mainnet fork", function () {
+        lpAmount1Min: 0,
+        lpLowerTick: -887220,
+        lpUpperTick: 887220
+      }
+
+      await controllerHelper.connect(depositor).flashloanWMintDepositNft(flashloanWMintDepositNftParams, {value: collateralToLp})
+    })
+
+    it("swap with pool, collect fees and redeposit uni nft in vault", async () => {
+      
+      const ethToSell = ethers.utils.parseUnits("500")
+      const swapParamBuy = {
+        tokenIn: weth.address,
+        tokenOut: wSqueeth.address,
+        fee: 3000,
+        recipient: owner.address,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+        amountIn: ethToSell.toString(),
+        amountOutMinimum: 0,
+        sqrtPriceLimitX96: 0
+      }    
+
+      weth.connect(owner).deposit({value: ethToSell})
+ 
+      const ownerSqueethBalanceBeforeTrade1 = await wSqueeth.balanceOf(owner.address)
+      const ownerWethBalanceBeforeTrade1 = await weth.balanceOf(owner.address)
+
+      await weth.connect(owner).approve(uniswapRouter.address, constants.MaxUint256)
+
+      await uniswapRouter.connect(owner).exactInputSingle(swapParamBuy)
+      const ownerSqueethBalanceAfterTrade1 = await wSqueeth.balanceOf(owner.address)
+      const ownerWethBalanceAfterTrade1 = await weth.balanceOf(owner.address)
+
+      expect(ownerWethBalanceBeforeTrade1.sub(ownerWethBalanceAfterTrade1).eq(ethToSell)).to.be.true
+
+      const wSqueethToSell = ownerSqueethBalanceAfterTrade1.sub(ownerSqueethBalanceBeforeTrade1)
+      const swapParamSell = {
+        tokenIn: wSqueeth.address,
+        tokenOut: weth.address,
+        fee: 3000,
+        recipient: owner.address,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+        amountIn: wSqueethToSell,
+        amountOutMinimum: 0,
+        sqrtPriceLimitX96: 0
+      }    
+
+      const ownerSqueethBalanceBeforeTrade2 = await wSqueeth.balanceOf(owner.address)
+      await wSqueeth.connect(owner).approve(uniswapRouter.address, constants.MaxUint256)
+      await uniswapRouter.connect(owner).exactInputSingle(swapParamSell)
+      const ownerSqueethBalanceAfterTrade2 = await wSqueeth.balanceOf(owner.address)
+      expect(ownerSqueethBalanceBeforeTrade2.sub(ownerSqueethBalanceAfterTrade2).eq(wSqueethToSell)).to.be.true
+
+      const vaultId = (await shortSqueeth.nextId()).sub(1);
+      const uniTokenId =  (await controller.vaults(vaultId)).NftCollateralId;
+      const vaultBefore = await controller.vaults(vaultId); 
+      const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+      const scaledEthPrice = ethPrice.div(10000)
+      const debtInEth = vaultBefore.shortAmount.mul(scaledEthPrice).div(one)
+      const collateralToFlashloan = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+      const positionBefore = await (positionManager as INonfungiblePositionManager).positions(uniTokenId);
+
+      const amount0Max = BigNumber.from(2).mul(BigNumber.from(10).pow(18)).sub(1)
```

Comment:
why not set amount0Max and amount1Max to uint256.max?

Reply:
I'm not sure why a user wouldn't want to collect all the fees, but didn't see a reason to restrict it. Figured just have the f/e pass it in, unless theres a reason I'm missing.